src/persist-client/src/impl/state.rs

```diff
@@ -75,6 +84,20 @@ where
         Continue((Upper(self.trace.upper().clone()), read_cap))
     }
 
+    pub fn register_writer(
+        &mut self,
+        writer_id: &WriterId,
+    ) -> ControlFlow<Infallible, (Upper<T>, WriterState)> {
+        let writer_state = WriterState {
+            // WIP(phemberger): convert to NowFn so we can swap this out for tests
+            //   and default to EPOCH_MILLIS at runtime. should the NowFn live on
+            //   StateCollections?
```

Comment:
really not sure the best way to do this. maybe `NowFn` could be a private field on `StateCollections`, and we have some method `with_now` that lets someone manually override if needed? or maybe it's set at some PersistConfig level? anything I can think of feels super invasive

Reply:
oh 100% the EpochMillis should just be passed in directly to the method on machine and threaded down here. it's fine for WriteHandle to call the real mz_ore::now::system_time() impl to fill this in, we'll do our correctness testing of it at the machine level